### PR TITLE
Bump `immer` to `9.0.6`

### PR DESCRIPTION
**Pull request by dimas-b**
This is mostly to bump the `immer` transitive dependency
to v9.0.6 to resolve CVE-2021-23436 and CVE-2021-3757.

----

Related: https://github.com/projectnessie/nessie/security/dependabot/ui/package-lock.json/immer/open

### CODE REVIEW

**Review by nastra — APPROVED** [review]

**Comment by ajantha-bhat** [comment]
@dimas-b : But dependabot says even 11.0.4 is mapped to V8.0.1 of immer,  but CVE fixed in 9.0.6 of immer.
Is dependabot wrong ? 🤔
Screenshot 2021-11-18 at 11 06 44 PM

File: ui/package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "react": "^16.14.0",
     "react-app-polyfill": "^2.0.0",
     "react-bootstrap": "^2.0.2",
-    "react-dev-utils": "^11.0.3",
+    "react-dev-utils": "^11.0.4",
```
Comment (ajantha-bhat) [inline ui/package.json:44]: https://github.com/facebook/create-react-app/blob/v4.0.3/packages/react-dev-utils/package.json

Comment (ajantha-bhat) [reply ui/package.json:44]: Issue is that `react-dev-utils` master branch code is pointing to  immer:9.0.6

but latest release is still pointing to "immer": "8.0.1".

So, I don't think this fix will solve the problem. we need to wait for new release.

**Review by ajantha-bhat — CHANGES_REQUESTED** [review]

File: ui/package.json (hunk 1 above)
Comment (ajantha-bhat) [reply ui/package.json:44]: dependabot also saying the same thing.

image

Comment (dimas-b) [reply ui/package.json:44]: you're right... I check `react-dev-utils` sources, but missed that the `9.0.6` upgrade was not released yet :facepalm:

Comment (ajantha-bhat) [reply ui/package.json:44]: cool. Let's wait for the new release. which dependabot will auto update when happens. we can sit back and relax !

Comment (dimas-b) [reply ui/package.json:44]: however https://github.com/facebook/create-react-app/blob/main/packages/react-dev-utils/package.json shows `immer` at `9.0.6` in `react-dev-utils` `11.0.4` :thinking:

Comment (dimas-b) [reply ui/package.json:44]: Should we try v. `12.0.0-next.47` for `react-dev-utils`?

Comment (ajantha-bhat) [reply ui/package.json:44]: True. Confusing. I checked from last release tag which is mapping to 11.0.3. 
not sure how 11.0.4 is out if version is still in master branch !

Comment (ajantha-bhat) [reply ui/package.json:44]: Are the independently releasing packages without sync ?

Comment (dimas-b) [reply ui/package.json:44]: `11.0.4` shows available here: https://www.npmjs.com/package/react-dev-utils/v/11.0.4

Comment (dimas-b) [reply ui/package.json:44]: but a local build with 11.0.4 still pulls `immer` 8.0.1 :(

Comment (dimas-b) [reply ui/package.json:44]: however, `immer` is not pinned, it seems, let me try to bump it individually :hourglass:

Comment (ajantha-bhat) [reply ui/package.json:44]: > Should we try v. 12.0.0-next.47 for react-dev-utils?
I think not good to use unreleased version (snapshots)

may be we can raise issue in their github and get it clarified or get info about next release.

**Comment by dimas-b** [comment]
Note: `mvn quarkus:dev` does not start the UI after the most recent quarkus version bump. You can validate UI by building and running a Nessie docker image.

File: ui/package.json (hunk 1 above)
Comment (dimas-b) [reply ui/package.json:45]: Added a direct dependency on `immer` to force bump it to `9.0.6` (force pushed)

Comment (ajantha-bhat) [reply ui/package.json:45]: but will it break `react-dev-utils` ? we can test once, if it is fine. we can merge it.

Comment (dimas-b) [reply ui/package.json:45]: I smoke-tested the UI in a local build - it was fine, but I'm fairly new to the whole UI tools chain, so I'd appreciate if someone could double check.

File: ui/package-lock.json [hunk 2]
```diff
@@ -38210,6 +38220,11 @@
             "slash": "^3.0.0"
           }
         },
+        "immer": {
+          "version": "8.0.1",
```
Comment (ajantha-bhat) [inline ui/package-lock.json:38224]: 8.0.1 has CVE, so as long as we are keeping it in the repo, we are vulnerable right ?

File: ui/package.json (hunk 1 above)
Comment (dimas-b) [reply ui/package.json:45]: > but will it break react-dev-utils ?

`create-react-app` apparently uses this version combination on `main`.

https://github.com/facebook/create-react-app/blob/main/packages/react-dev-utils/package.json

File: ui/package-lock.json (hunk 2 above)
Comment (dimas-b) [reply ui/package-lock.json:38224]: TBH, I do not know how it works. This is a nested dep. from `react-dev-utils`. The top-level entry for `immer` is at `9.0.6` now.

Comment (dimas-b) [reply ui/package-lock.json:38224]: Is there a way to verify what is actually packaged?

Comment (ajantha-bhat) [reply ui/package-lock.json:38224]: I too have no idea for this side of the code. Theoretically the current changes looks ok to me. But better to consult UI experts :)

**Review by ajantha-bhat — COMMENTED** [review]
Theoretically the current changes looks ok to me compared to last one. But better to consult UI experts :)

**Review by ajantha-bhat — APPROVED** [review]

File: ui/package.json (hunk 1 above)
Comment (dimas-b) [reply ui/package.json:44]: It turned out to be complicated... I did another force push to use `npm-force-resolutions` for bumping nested `immer` dependencies to `9.0.6`.

File: ui/package-lock.json (hunk 2 above)
Comment (dimas-b) [reply ui/package-lock.json]: It turned out to be complicated... I did another force push to use `npm-force-resolutions` for bumping nested `immer` dependencies to `9.0.6`.

Comment (ajantha-bhat) [reply ui/package-lock.json]: It shows 34KLOC diff, do we really need to handle this way ? I think we can consult `react-dev-utils` and ask them for a formal release ?

**Comment by dimas-b** [comment]
Let's wait for https://github.com/facebook/create-react-app/issues/11660 ... closing

**Pull request closed without merging**